src/cockpit/channels/dbus.py

```diff
@@ -282,7 +282,15 @@ def filter_owner(message):
             func = handler
         r_string = ','.join(f"{key}='{value}'" for key, value in r.items())
         if not self.is_closing():
-            self.matches.append(self.bus.add_match(r_string, func))
+            # this gets an EINTR very often especially on RHEL 8
```

Comment:
I wonder if we can do some root cause analysis here.  This might only be happening in a particular situation which we might avoid.  As far as I can tell, sd-bus never returns `EINTR` on its own, which means that this is most probably coming from some lower-level syscall returning `EINTR` which could then pop up as a return value in response to other calls....

An alternative might also be to try and deal with this at the binding level.  We already have a handler that inspects the return values of these functions and converts them into `OSError`.  Maybe we could redo the call in case we get EINTR?

Reply:
ps: relevant systemd commits look like b3d06b9226db96fddb6bb45a4708e2e8d413d91d and 3022916b4d2483452c3ddbbac9ee7c4372b1cb46.  It seems that they have an `ERRNO_IS_TRANSIENT()` macro to deal with fails like these and have been increasing its use recently.  I wonder if it makes sense to try and backport those patches into RHEL?

Reply:
I actually ran into the EINTR with `journalctl` during my interactive tests. As it happens, this is already being backported in https://bugzilla.redhat.com/show_bug.cgi?id=2161683 . But this is completely independent from the sd-bus issue.

I reported this in https://bugzilla.redhat.com/show_bug.cgi?id=2172846 and asked for backporting this. If/once that happens, we can remove the workaround.

I looked into systemd_ctypes, and I didn't find a general wrapper around all bindings which would be able to run the function multiple times. There is `negative_errno()` (what you meant with "convert them into OSError"), but that does not have any context to be able to re-run the call. Possibly we could work this into `def instancemethod()`'s `wrapper()`? What would be the equivalent wrapper for `staticmethod`?

Reply:
@allisonkarlitskaya : Forgot to tag you for my previous question :arrow_up: